src/Microsoft.AspNet.Mvc.Core/Rendering/Html/TemplateRenderer.cs

```diff
@@ -168,6 +171,14 @@ private IEnumerable<string> GetViewNames()
 
                 yield return "String";
             }
+            else if (typeof(IFormFile).IsAssignableFrom(fieldType))
+            {
+                yield return nameof(IFormFile);
+            }
+            else if (typeof(IEnumerable<IFormFile>).IsAssignableFrom(fieldType))
+            {
+                yield return nameof(IFormFileCollection);
+            }
```

Comment:
this gives the interfaces too high a priority.  fine for the `fieldType.IsInterface()` case but should for example yield `FormFile` before `IFormFile` if the property is of that type or a subtype.  need to add these checks immediately before where this method yields "Collection" and "Object".  for example,
``` c#
            else if (fieldType.IsInterface())
            {
                if (typeof(IEnumerable).IsAssignableFrom(fieldType))
                {
                    if (typeof(IEnumerable<IFormFile>).IsAssignableFrom(fieldType))
                    {
                        yield return nameof(IFormFileCollection);
                    }

                    yield return "Collection";
                }
            ...
```

will likely lead to a few duplicate lines but that's better than not returning the expected types.